name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/splits/StockSplitModel.java

```diff
@@ -93,22 +103,26 @@ public void setChangeHistoricalQuotes(boolean changeHistoricalQuotes)
                         this.changeHistoricalQuotes = changeHistoricalQuotes);
     }
 
-    // stock multiplier = quote divider
-    private BigDecimal getStockMultiplier()
+    private BigDecimal calculateStockMultiplier()
     {
         return newShares.divide(oldShares, Values.MC);
     }
     
+    private BigDecimal calculateQuoteMultiplier()
+    {
+        return oldShares.divide(newShares, Values.MC);
+    }
+    
     public long calculateNewStock(long oldStock)
     {
-        return BigDecimal.valueOf(oldStock).multiply(getStockMultiplier())
+        return BigDecimal.valueOf(oldStock).multiply(stockMultiplier)
                         .setScale(0, RoundingMode.HALF_EVEN).longValue();
     }
     
     public long calculateNewQuote(long oldQuote)
     {
-        return BigDecimal.valueOf(oldQuote).divide(getStockMultiplier()) // when stock is multiplied, quote must be divided
-                        .setScale(0, RoundingMode.HALF_EVEN).longValue();        
+        return BigDecimal.valueOf(oldQuote).multiply(quoteMultiplier)
```

Comment:
Do you really need to variables - stock and quote multipler - or couldn't you still call BigDecimal#divide but simply pass the math context?

Reply:
@buchen: Of course, using single field also works. I tried BigDecimal#divide(stockMultiplier, Values.MC) locally and the unittest succeeds.
Should I switch to single field?